include/mbedtls/ssl.h

```diff
@@ -4849,6 +4853,24 @@ int mbedtls_ssl_get_session(const mbedtls_ssl_context *ssl,
  *                 currently being processed might or might not contain further
  *                 DTLS records.
  *
+ * \note           In TLS, reception of fragmented handshake messages is
+ *                 supported with some limitations (those limitations do
+ *                 not apply to DTLS, where defragmentation is fully
+ *                 supported):
+ *                 - On an Mbed TLS server that only accepts TLS 1.2,
+ *                   the initial ClientHello message must not be fragmented.
+ *                   A TLS 1.2 ClientHello may be fragmented if the server
+ *                   also accepts TLS 1.3 connections (meaning
+ *                   that #MBEDTLS_SSL_PROTO_TLS1_3 enabled, and the
+ *                   accepted versions have not been restricted with
+ *                   mbedtls_ssl_conf_max_tls_version() or the like).
+ *                 - The first fragment of a handshake message must be
+ *                   at least 4 bytes long.
+ *                 - Non-handshake records must not be interleaved between
+ *                   the fragments of a handshake message. (This is permitted
+ *                   in TLS 1.2 but not in TLS 1.3, but Mbed TLS rejects it
+ *                   even in TLS 1.2.)
+ *
```

Comment:
Should we add an entry for the renegotiation issue that we have come accross?

> Users utilizing renegotiation with the library's default renegotiation delay are advised to increase it to 24/36 using `mbedtls_ssl_conf_renegotiation_enforced` if their application is expected to process small fragment sizes (<312 bytes )

It may save us from a future bug report of my server is not working with this client etc.

Reply:
I think renegotiaton is enough of an edge case that I'm ok leaving it out initially. We can always add a note later if actual users (as opposed to us testers) run into it.

Reply:
The documentation of `mbedtls_ssl_conf_renegotiation_enforced` already states “The optimal value is highly dependent on the specific usage scenario”, so I think we're ok.